lib/mapping/MapFormatJson.cpp

```diff
@@ -960,12 +960,7 @@ void CMapLoaderJson::readTerrainTile(const std::string & src, TerrainTile & tile
 	{//terrain type
 		const std::string typeCode = src.substr(0, 2);
 
-		int rawType = vstd::find_pos(terrainCodes, typeCode);
-
-		if(rawType < 0)
-			throw std::runtime_error("Invalid terrain type code in "+src);
-
-		tile.terType = ETerrainType(rawType);
+		tile.terType = Terrain(src);

```

Comment:
- `typeCode` is defined in  `namespace TerrainDetail` in this file. It should be moved to terrains.json config and handled somehow.
- `src` is not even a terrain name, it 2 letter terrain code + "terrain view" + "terrain flip"

Reply:
fixed